src/main/java/org/opentripplanner/middleware/controllers/api/ApiUserController.java

```diff
@@ -84,20 +96,39 @@ private boolean userHasKey(ApiUser user, String apiKeyId) {
                 .anyMatch(apiKey -> apiKeyId.equals(apiKey.keyId));
     }
 
+    /**
+     * Authenticate user with Auth0 based on username (email) and password. If successful, return the complete Auth0
+     * token else log message and halt.
+     */
+    private TokenHolder authenticateAuth0User(Request req, Response res) {
+        String username = HttpUtils.getQueryParamFromRequest(req, USERNAME_PARAM, false);
+        // FIXME: Should this be encrypted?!
+        String password = HttpUtils.getQueryParamFromRequest(req, PASSWORD_PARAM, false);
+        HttpResponse<String> auth0TokenResponse = Auth0Users.getCompleteAuth0TokenResponse(username, password);
+        if (auth0TokenResponse == null || auth0TokenResponse.statusCode() != HttpStatus.OK_200) {
```

Comment:
Should be a check against `x-api-key` when using the `/authenticate/` endpoint? As written, you could obtain a token using any key as long as it is recognized by API gateway (or you could use no key if not behind an API gateway, and still get a token).

Reply:
Updated.